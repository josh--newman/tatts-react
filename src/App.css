body {
   font-family: "HelveticaNeue-Light", "Helvetica Neue Light", "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;
   font-weight: 300;
}

.container {
  text-align: center;
}

.list {
  list-style: none;
  margin: 0 auto;
  padding: 0;
  width: 500px;
  display: flex;
  flex-direction: column;
}

.list-item {
  border-bottom: 1px solid lightgrey;
  display: flex;
  align-items: center;
}

.logo {
  width: 150px;
  max-height: 150px;
  overflow: hidden;
  margin-right: 10px;
}

.arrow {
  margin-left: auto;
}
