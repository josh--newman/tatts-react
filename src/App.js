import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './App.css';
import { fetchCompanies } from './api';
import Company from './Company';

class App extends Component {
  state = { companies: [], loading: false };

  componentDidMount() {
    this.setState({ loading: true });
    return this.props._fetchCompanies()
      .then(companies => this.setState({ companies, loading: false }))
      .catch(err => {
        this.setState({ loading: false });
        throw new Error(err);
      });
  }

  render() {
    if (this.state.loading) {
      return <p>Loading...</p>;
    }
    else {
      return (
        <div className="container">
          <img src="https://thelott.com/etc/designs/the-lott/components/structure/page-template/clientlibs/images/header/logo--dsktp-thelott-australias.png"
          alt="The Lott logo" title="the Lott, Australia's Official Lotteries" />
          <p>Please select your lottery provider</p>
          <ul className="list">
            {
              this.state.companies.map(company => {
                return <Company key={company.id} {...company} />
              })
            }
          </ul>
        </div>
      )
    }
  }
}

App.propTypes = {
  loading: PropTypes.bool.isRequired,
  companies: PropTypes.array.isRequired
}

App.defaultProps = {
  loading: false,
  companies: [],
  _fetchCompanies: fetchCompanies
}

export default App;
