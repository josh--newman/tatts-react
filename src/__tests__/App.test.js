import React from 'react';
import { shallow } from 'enzyme';
import { spy } from 'sinon';
import App from '../App';

const companies = [
  {
    id: "GoldenCasket",
    displayName: "Golden Casket",
    description: "QLD Residents",
    logoUrl: "http://tim.media.tatts.com/TattsServices/Lotto/Companies/GoldenCasket_v1.png"
  },
  {
    id: "NSWLotteries",
    displayName: "NSW Lotteries",
    description: "NSW Residents",
    logoUrl: "http://tim.media.tatts.com/TattsServices/Lotto/Companies/NSWLotteries_v1.png"
  },
];

const baseProps = {
  _fetchCompanies: () => Promise.resolve(companies)
}

describe('App', function() {
  it('renders', function() {
    const component = shallow(<App {...baseProps} />);
    expect(component).toMatchSnapshot();
  });

  it('fetches companies', function() {
    const fetchCompanies = jest.fn(() => Promise.resolve(companies));
    const component = shallow(
      <App _fetchCompanies={fetchCompanies} />
    );
    expect(fetchCompanies).toHaveBeenCalled();
  });
});
