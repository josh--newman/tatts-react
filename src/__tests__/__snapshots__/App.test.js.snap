// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    _fetchCompanies={[Function]}
    companies={Array []}
    loading={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": "Loading...",
    },
    "ref": null,
    "rendered": "Loading...",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "Loading...",
      },
      "ref": null,
      "rendered": "Loading...",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
